library/Graphite/Graphing/GraphiteWebClient.php

```diff
@@ -94,13 +94,19 @@ public function request(Url $url, $method = 'GET', array $headers = [], $body =
      */
     public function escapeMetricPath($metricPath)
     {
-        return preg_replace_callback(
+        // TODO: Don't know what is the reason behind this?
+        // it changes '.u[sd]ers.value' into '.u[[]sd[]].value'
+        // and cause empty result at the end
```

Comment:
I don't either. This was introduced by 839ca7e6185f9e5bd10c5d7e95b1a4f5f734d27b. @Al2Klimov any hint?

Reply:
Graphite Web takes patterns, not literal strings. That was intended to make [ ] working in metric names.

Reply:
I see. Though we cannot remove the method. It's just applied incorrectly. It cannot be used to escape the entire metric path, only selected parts. I suppose it's sufficient to escape the results of macros. (i.e. The ones inside `$host_name_template$` and `$service_name_template$`) Additionally, the documentation should describe how to escape `[ ]` and `{ }` in metric filters of templates.

So that this works:

host.name = foo[bar]
host.check_command = rab[oof]
host_name_template = icinga2.$host.name$.host.$host.check_command$
users = $host_name_template$.perfdata.us[a-z]rs.value
result = icinga2.foo[[]bar[]].host.rab[[]oof[]].perfdata.us[a-z]rs.value